SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebuggerEditor.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialDebuggerEditor.h"
+
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include "LoadBalancing/LayeredLBStrategy.h"
+#include "SpatialCommonTypes.h"
+#include "Utils/InspectionColors.h"
+
+#include "Debug/DebugDrawService.h"
+#include "Engine/World.h"
+#include "GameFramework/WorldSettings.h"
+#include "Modules/ModuleManager.h"
+
+using namespace SpatialGDK;
+
+ASpatialDebuggerEditor::ASpatialDebuggerEditor(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+{
+	PrimaryActorTick.bCanEverTick = false;
+	bReplicates = false;
+}
+
+void ASpatialDebuggerEditor::Destroyed()
+{
+	DestroyWorkerRegions();
+}
+
+void ASpatialDebuggerEditor::ToggleWorkerRegionVisibility(bool bEnabled)
+{
+	bShowWorkerRegions = bEnabled;
+	RefreshWorkerRegions();
+}
+
+void ASpatialDebuggerEditor::RefreshWorkerRegions()
+{
+	DestroyWorkerRegions();
+
+	if (bShowWorkerRegions && AllowWorkerBoundaries())
+	{
+		InitialiseWorkerRegions();
+		CreateWorkerRegions();
+	}
+#if WITH_EDITOR
+	if (GEditor != nullptr && GEditor->GetActiveViewport() != nullptr)
+	{
+		// Redraw editor window to show changes
+		GEditor->GetActiveViewport()->Invalidate();
+	}
+#endif
+}
+
+bool ASpatialDebuggerEditor::AllowWorkerBoundaries() const
+{
+#if WITH_EDITOR
+	// Check if multi worker is enabled.
+	UWorld* World = GetWorld();
+
+	check(World != nullptr);
+		
+	const ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(World->GetWorldSettings());
+	const bool bIsMultiWorkerEnabled = WorldSettings != nullptr && WorldSettings->IsMultiWorkerEnabled();
```

Comment:
I think we want to use `USpatialStatics::IsSpatialMultiWorkerEnabled` here instead

Reply:
After the merge with Ally's changes this got updated to: 
`const bool bIsSpatialNetworkingEnabled = GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking();`

I tried to use` USpatialStatics::IsSpatialMultiWorkerEnabled` but it does not update when you toggle the dropdown "SpatialOS Networking" on the "Start Deployment" drop-down menu and fails the tests.

Reply:
My mistake I changed the wrong line. Now I have updated to `USpatialStatics::IsSpatialMultiWorkerEnabled` and it is all working.